pkg/runtime/hotreload/reconciler/component.go

```diff
@@ -48,32 +53,40 @@ func (c *component) update(ctx context.Context, comp componentsapi.Component) {
 	if exists {
 		if differ.AreSame(oldComp, comp) {
 			log.Debugf("Component update skipped: no changes detected: %s", comp.LogName())
-			return
+			return nil
 		}
 
 		log.Infof("Closing existing Component to reload: %s", oldComp.LogName())
 		// TODO: change close to accept pointer
 		if err := c.proc.Close(oldComp); err != nil {
 			log.Errorf("error closing old component: %w", err)
-			return
+			return nil
 		}
 	}
 
 	if !c.auth.IsObjectAuthorized(comp) {
 		log.Warnf("Received unauthorized component update, ignored: %s", comp.LogName())
-		return
+		return nil
 	}
 
 	log.Infof("Adding Component for processing: %s", comp.LogName())
 	if c.proc.AddPendingComponent(ctx, comp) {
 		log.Infof("Component updated: %s", comp.LogName())
 		c.proc.WaitForEmptyComponentQueue()
 	}
+
+	return nil
 }
 
 //nolint:unused
-func (c *component) delete(comp componentsapi.Component) {
+func (c *component) delete(comp componentsapi.Component) error {
+	if strings.HasPrefix(comp.Spec.Type, "workflowbackend.") {
+		return fmt.Errorf("attempting to HotReload a workflowbackend component which is not supported: %s", comp.LogName())
+	}
+
 	if err := c.proc.Close(comp); err != nil {
 		log.Errorf("error closing deleted component: %w", err)
```

Comment:
Logs don't support `%w`
```suggestion
		log.Errorf("error closing deleted component: %v", err)
```